File: app/src/main/java/gb/android/notes2/view/ViewManager.java
package gb.android.notes2.view;

import android.content.res.Configuration;

import gb.android.notes2.MainActivity;
import gb.android.notes2.R;
import gb.android.notes2.view.notelist.NoteListAdapter;
import observer.Publisher;

public class ViewManager {
    private static MainActivity mainActivity;

    private static NoteListAdapter noteListAdapter;

    private static Publisher publisher;

    //===============================================================================================
    // STATIC GETTERS

    static public MainActivity getMainActivity() {
        return mainActivity;
    }

    static public NoteListAdapter getNoteListAdapter() {
        return noteListAdapter;
    }

    static public Publisher getPublisher() {
        if (publisher == null)
            publisher = new Publisher();
        return publisher;
    }

    static public int getScreenOrientation() {
        return mainActivity.getResources().getConfiguration().orientation;
    }

    //===============================================================================================
    // STATIC SETTERS

    static public void setMainActivity(MainActivity mainActivity) {
        ViewManager.mainActivity = mainActivity;
    }

    static public void setNoteListAdapter(NoteListAdapter noteListAdapter) {
        ViewManager.noteListAdapter = noteListAdapter;
    }

    //===============================================================================================
    // FRAGMENTS

    static public void startEditor() {
        if (getScreenOrientation() == Configuration.ORIENTATION_PORTRAIT) {
            mainActivity.getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.container, NoteEditorFragment.newInstance())
                    .addToBackStack("")
                    .commit();
        } else {
            mainActivity.getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.container_right, NoteEditorFragment.newInstance())
                    .addToBackStack("")
                    .commit();
        }
    }

}

File: app/src/main/java/gb/android/notes2/view/ViewManager.kt
package gb.android.notes2.view

import android.content.res.Configuration
import gb.android.notes2.MainActivity
import gb.android.notes2.view.notelist.NoteListAdapter
import gb.android.notes2.R
import gb.android.notes2.observer.Publisher
import gb.android.notes2.view.NoteEditorFragment

object ViewManager {
    //===============================================================================================
    // STATIC SETTERS
    //===============================================================================================
    // STATIC GETTERS
    @JvmStatic
    var mainActivity: MainActivity? = null
    @JvmStatic
    var noteListAdapter: NoteListAdapter? = null
    @JvmStatic
    var publisher: Publisher? = null
        get() {
            if (field == null) field = Publisher()
            return field
        }
        private set
    @JvmStatic
    val screenOrientation: Int
        get() = mainActivity!!.resources.configuration.orientation

    //===============================================================================================
    // FRAGMENTS
    @JvmStatic
    fun startEditor() {
        if (screenOrientation == Configuration.ORIENTATION_PORTRAIT) {
            mainActivity!!.supportFragmentManager
                .beginTransaction()
                .replace(R.id.container, NoteEditorFragment.newInstance())
                .addToBackStack("")
                .commit()
        } else {
            mainActivity!!.supportFragmentManager
                .beginTransaction()
                .replace(R.id.container_right, NoteEditorFragment.newInstance())
                .addToBackStack("")
                .commit()
        }
    }
}